test/WebJobs.Script.Tests/Rpc/GenericWorkerProviderTests.cs

```diff
@@ -180,7 +207,39 @@ private IEnumerable<IWorkerProvider> TestReadWorkerProviderFromConfig(string lan
                 };
                 scriptSettingsManager.SetConfigurationFactory(() => new ConfigurationBuilder()
                     .AddInMemoryCollection(settings).Build());
-                return GenericWorkerProvider.ReadWorkerProviderFromConfig(scriptHostConfig, mockLogger.Object, scriptSettingsManager);
+                return (workerPath, GenericWorkerProvider.ReadWorkerProviderFromConfig(scriptHostConfig, mockLogger.Object, scriptSettingsManager));
+            }
+            finally
+            {
+                if (Directory.Exists(rootPath))
+                {
+                    Directory.Delete(rootPath, true);
+                }
+            }
+        }
+
+        private (string workerPath, IEnumerable<IWorkerProvider> providers) TestReadWorkerProviderFromConfig(IEnumerable<TestLanguageWorkerConfig> configs, Mock<ILogger<object>> mockLogger, string language = null)
+        {
+            string rootPath = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+            try
+            {
+                foreach (var config in configs)
+                {
+                    string workerPath = Path.Combine(rootPath, config.Language);
+                    Directory.CreateDirectory(workerPath);
+
+                    File.WriteAllText(Path.Combine(workerPath, ScriptConstants.WorkerConfigFileName), config.Json);
+                }
+
+                var scriptHostConfig = new ScriptHostConfiguration();
+                var scriptSettingsManager = new ScriptSettingsManager();
+                var settings = new List<KeyValuePair<string, string>>
```

Comment:
`Dictionary<string, string>` ?

Reply:
`.AddInMemoryCollection(settings)` wants a list of KeyValuePairs, not a dictionary.

Reply:
That's not correct... :)
`.AddInMemoryCollection(settings)` takes an `IEnumerable<KeyValuePair<string, string>>`, which a `Dictionary<string, string>` is